module-2/backend/week_three.md

```diff
@@ -10,14 +10,41 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+   rails new project_name
+   
 2. What do Models generally inherit from in rails?
+  ActiveRecord::Base
+  
 3. What do Controllers generally inherit from in a rails project?
+  ApplicationController
+  
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  resources :horses, only: [:show]
+  
 5. What rake task is useful when looking at routes, and what information does it give you?
+  rake routes
+  
 6. What is an example of a route helper? When would you use them?
+  new_category_path - used anytime you need to specify a location (i.e. link_to, redirect_to, etc)
+  
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+  ???
+  
 8. What are strong params and why are the necessary?
+  params.require().permit() - something about mass assignment and security but can't say exactly
+  
 9. What role does `form_for` play in helping us create our forms?
+  denotes a form and lets us set/iterate through each field (f.label :column_name)
+  
 10. How does `form_for` know where to submit the user's input?
+  the path is set in the first line of the form (form_for @horse) which is then traced to the controller to choose the correct method   depending on the path/verb/view it's being sent from (@horse in the create method for 'new' or @horse in the update method     for 'edit')
```

Comment:
close. It checks if the instance variable (@horse) has been created before the `new_record?` method. If it is new it will send the form to post if it has been in the database before it will send the form to put/patch